### PR TITLE
Convert project to maven and adds the launcher project

### PR DESCRIPTION
**Pull request by brownan**
This pull request is the squashed changes from everything I've done. The commit message is below.

---

This commit shuffles around source files and adds maven pom.xml files with
dependencies.

The ScreenCastingLocalHub project was moved to a subproject, and the launcher
was added as a new subproject. The top level maven pom.xml is the parent of the
two projects.

The dependency management provided my Maven makes the externalJars
directory obsolete and it can probably be removed, but it is left in case I
missed something. (Not every jar in externalJars was turned in to a maven
dependency because not all appear to be used)

Note that a local maven repository was added in the ScreenCastingLocalHub/repo
directory to hold the sslcerts.jar dependency.

This commit also copies in the ScreenRecordingModule.java file and its
supporting files from the other repository.

Also note that there was an old "launcher" in the launcher directory. It has
been removed.

And finally, this adds a travis config for continuous integration support.

### CODE REVIEW

File: ScreenCastingLocalHub/pom.xml [hunk 1]
```diff
@@ -0,0 +1,125 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>edu.ncsu.lubick</groupId>
```
Comment (kjlubick) [inline ScreenCastingLocalHub/pom.xml]: If it's not too much hassle, can you change the group id to be edu.ncsu.dlf?  Basically, find replace lubick with dlf

Comment (brownan) [inline ScreenCastingLocalHub/pom.xml]: sure. let's see what IntelliJ's refactoring tools can do!

**Comment by brownan** [comment]
@kjlubick Added another commit to this pull request, changing the package edu.ncsu.lubick to edu.ncsu.dlf.

Oops I forgot to test it 1 sec...

**Comment by brownan** [comment]
okay it seems to work okay. nothing went wrong with the refactoring.

**Pull request merged**